config/v1/types_ingress.go

```diff
@@ -41,13 +44,103 @@ type IngressSpec struct {
 	// certificate.
 	// +optional
 	AppsDomain string `json:"appsDomain,omitempty"`
+
+	// ComponentRoutes is a list of routes configured by a cluster-admin.
+	// The namespace and name of each componentRoute should match an existing Route.
+	//
+	// To determine the set of configurable Routes, look at namespace and name of entries in the
+	// .status.componentRoutes, where participating operators place current route status.
+	// +kubebuilder:validation:Optional
```

Comment:
I think my previous comment was wrong. Compare https://github.com/openshift/api/blob/master/config/v1/doc.go#L5. We have optional by default for kubebuilder. So you only have to specify required fields. On the other hand, this does not harm either.

Reply:
You are correct that all items default to optional. I removed all `// +kubebuilder:validation:Optional` instances.